### ISSUE TITLE
New go at listLengthRange (on behalf of @svarlet)

### ISSUE THREAD

**Issue by mrbackend** [elm-community/elm-test#111]
There was a discussion in the #testing channel on Slack yesterday, where @rtfeldman moved to open a new issue about this old candidate. I'm not the one who asked for it on the Slack channel , but I have a pretty straightforward implementation that I wanted to suggest:

```
import Fuzz exposing (Fuzzer)


{-| A fuzzer for lists with a length within the given range (inclusive).

If maxLength < minLength, the result is undefined.
-}
listLengthRange : Int -> Int -> Fuzzer a -> Fuzzer (List a)
listLengthRange minLength maxLength elemFuzzer =
    Fuzz.intRange minLength maxLength
        |> Fuzz.andThen (combine << flip List.repeat elemFuzzer)


{-| traverse and combine would like to be in the Fuzz module anyway
-}
traverse : (a -> Fuzzer b) -> List a -> Fuzzer (List b)
traverse f =
    List.foldr (Fuzz.map2 (::) << f) (Fuzz.constant [])


combine : List (Fuzzer a) -> Fuzzer (List a)
combine =
    traverse identity
```

**Comment by rtfeldman** [comment]
Reproducing the motivating conversation from Slack:

> Can anyone confirm that there is currently no option to limit the number of generated elements for a list fuzzer ?
> *(someone asked):* wouldn't `Fuzz.map (List.take 42)` work?
> no because I’d like to shorten the time it takes to generate the data
> I know my list will be at best 4 elements long
> there’s no point in letting the fuzzer generate a list of 1000 element
> it just takes a lot longer
> it becomes [a bottleneck] if you compose fuzzers
> such as a list of lists
> multiplied by the number of runs of this specific test
> my test suite went from 75ms to 15 seconds after adding such a test

tl;dr this is not necessarily a performance bottleneck when you want a `Fuzzer (List a)`, but when you want a `Fuzzer (List (List a))` the slowness gets compounded to the point where the performance difference can be severe.

**Comment by mgold** [comment]
> I know my list will be at best 4 elements long

Can you do `Fuzz.map4` on a fuzzer of Maybes, repeated for times, stuffed into a list and passed to `List.filterMap identity`? True you will only get 4 things one in 16 times; perhaps `Fuzz.frequency` could help with that.

**Comment by mgold** [comment]
Stale.
